solidity-checks/src/main/java/org/sonarsource/solidity/checks/FileRuleVerifier.java

```diff
@@ -0,0 +1,80 @@
+package org.sonarsource.solidity.checks;
+
+import com.sonarsource.checks.verifier.SingleFileVerifier;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import org.antlr.v4.runtime.CharStream;
+import org.antlr.v4.runtime.CharStreams;
+import org.antlr.v4.runtime.Token;
+import org.sonar.api.utils.log.Logger;
+import org.sonar.api.utils.log.Loggers;
+import org.sonarsource.solidity.frontend.SolidityParser;
+import org.sonarsource.solidity.frontend.Utils;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class FileRuleVerifier {
+
+  private static final Logger LOG = Loggers.get(FileRuleVerifier.class);
+
+  private static void verifyIssueOnFile(IssuableVisitor checkVisitor, String relativePath, String reportMessage) {
+    File file = new File(relativePath);
+    SingleFileVerifier verifier = SingleFileVerifier.create(file.toPath(), UTF_8);
+    CharStream cs;
+    try {
+      cs = CharStreams.fromFileName(relativePath);
+      SolidityParser parser = Utils.returnParserFromParsedFile(cs);
+      TestRuleContext testRuleContext = new TestRuleContext(verifier);
+      checkVisitor.setRuleContext(testRuleContext);
+      checkVisitor.visit(parser.sourceUnit());
+      if (reportMessage != null) {
+        assertThat(testRuleContext.issues).isNotEmpty();
+        assertThat(testRuleContext.issues).hasSize(1);
+        assertThat(testRuleContext.issues.get(0)).isEqualTo(reportMessage);
+      }
+    } catch (IOException e) {
+      LOG.debug(e.getMessage(), e);
+    }
+
+  }
+
+  public FileRuleVerifier(IssuableVisitor check, String relativePath, String reportMessage) {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Hide this public constructor. ([squid:S1118](https://sonarcloud.io/organizations/stylianos-agapiou-sonarsource-gi/rules#rule_key=squid%3AS1118))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=org.sonarsource.solidity%3Asonar-solidity&pullRequest=41&issues=AWSJ8a2_l6vpliCeT5VC&open=AWSJ8a2_l6vpliCeT5VC)

[ISSUE_KEY]: AWSJ8a2_l6vpliCeT5VC